src/main/scala/com/microsoft/azure/cosmosdb/spark/CosmosDBConnection.scala

```diff
@@ -228,6 +228,12 @@ private[spark] case class CosmosDBConnection(config: Config) extends LoggingTrai
       case Some(maxPoolSizeStr) => connectionPolicy.setMaxPoolSize(maxPoolSizeStr.toInt)
       case None => // skip
     }
+
+    config.get[String](CosmosDBConfig.ConnectionRequestTimeout) match {
+      case Some(connectionRequestTimeoutStr) => connectionPolicy.setRequestTimeout(connectionRequestTimeoutStr.toInt)
```

Comment:
shouldn't we use that for the async connection too ?

Reply:
True - added - covered for difference in time unit between sync and async SDK too